### DISCUSSION TITLE
Contributing guidelines

### DISCUSSION THREAD

**Discussion by gkieszkowski** [nasa/fprime#948] [General]
Hi,
I would like to start discussion on contributing guidelines to this repository. 
It would be much easier for newcomers to contribute if we had some short guidelines.
Also it would checkout the checklist boxes in [insights/community](https://github.com/nasa/fprime/community) tab and could benefit as potential contributors would see project as more attractive and be more likely to be involved in.
Mainly it should provide information on:
- communication guidelines
- issue template 
- branch naming
- workflow overview (description of branches, PR and merging process, code review process)
- issue/ PR labels
- commit messages guidelines
- list of maintainers and information who to mention regarding what topic (and how)
- navigating through issues and discussions
- reporting bugs
- suggesting new features
- where to start

In general contributing guidelines could set some standard to help us communicate better and add value to the project faster.
Would like to get your thoughts and suggestions on that matter.
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

**Comment by LeStarch** [comment]
I agree completely that we need this.

**Reply by ThibFrgsGmz** [reply]
I notice that the ``good first issue`` label has been removed from the list of labels.

Maybe it would be interesting to add it back?

Normally this is the label under which easily selectable issues of a project are collected, so that developers who have never contributed to an open source project can get started quickly.

The project maintainers are open to external contributors and are always looking to involve more people, but new developers probably think it's hard to become a contributor, and rightly so.

Perhaps adding this label and tagging the easy-to-implement issues would make it easier for developers to contribute?
Personally I think it would help me to contribute to the project.

**Comment by gkieszkowski** [comment]
Hi,
I have prepared [PR](https://github.com/nasa/fprime/pull/1046) (draft) of proposal of contributing guidelines.
It is very basic one but I figured out that it will be easier to work in that.
Currently it needs information on branch naming and commit messaging naming.

In general contributing guidelines are a way for contributors to get up to speed easier and for maintainers to provide some standards. We can all benefit from that and that's why I would love to ask you guys to edit this file as some of contributors here have much better experience and knowledge on nasa/fprime.

Hope we can create some minimal guidelines.

Also Iagree with `good first issue` label.

**Comment by ThibFrgsGmz** [comment]
According to the most recent GiHub octoverse [article](https://octoverse.github.com/creating-documentation/#good-first-issues) on "Creating documentation to support developers," "good first issues" guide new members' contributions.


According to their survey, repositories where 21% to 30% of publications are labeled as Good First Issues have an average of 13-15% new contributors. Furthermore, having more than 40% Good First Issues can help a project attract 21% more new contributors.

To accomplish this, GitHub recommends creating a set of tasks/issues that new people can begin with when they join the project, team, or community.

**Comment by LeStarch** [comment]
Note: submitted a style guide.  Edits/clarifications appreciated:
https://github.com/nasa/fprime/pull/1197
